### PR TITLE
[IPU]update paddle.distributed.launch

### PR DESCRIPTION
**Pull request by yaozhixin**
### PR types
New features

### PR changes
APIs

### Describe
Update paddle.distributed.launch with IPUs based on feedback.

### CODE REVIEW

File: python/paddle/distributed/launch/utils/process_context.py [hunk 1]
```diff
@@ -36,7 +36,16 @@ def __init__(self,
 
     def _start(self):
         pre_fn = os.setsid if self._group else None
-        self._proc = subprocess.Popen(self._cmd,
+        # Need shell=True for IPU PopRun command
+        if "poprun" in self._cmd[0]:
```
Comment (kuizhiqing) [inline python/paddle/distributed/launch/utils/process_context.py]: Consider add self._shell default False for process_context, add it to container too. 
Then handle it in controller with pod.

Let's keep process_context in low level and independent.

**Review by kuizhiqing — DISMISSED** [review]
LGTM

**Review by qili93 — DISMISSED** [review]
LGTM

**Review by kuizhiqing — DISMISSED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**